shared/router-v2/tab-bar/index.desktop.js

```diff
@@ -31,75 +31,136 @@ const tabs = [
-              fullWidth={true}
-              style={styles.nameContainer}
-              alignItems="center"
-            >
-              <Kb.Avatar
-                size={24}
-                borderColor={Styles.globalColors.blue}
-                username={p.username}
-                style={styles.avatar}
-              />
-              <Kb.Text className="username" type="BodyTinySemibold" style={styles.username}>
-                Hi {p.username}!
-              </Kb.Text>
-              <Kb.Icon
-                type="iconfont-arrow-down"
-                color={Styles.globalColors.blue3}
-                fontSize={12}
-                style={styles.caret}
-              />
-            </Kb.Box2>
-          </Kb.ClickableBox>
-          <Kb.Divider style={styles.divider} />
-        </Kb.Box2>
-        {tabs.map(t => (
-          <Kb.ClickableBox key={t} onClick={() => p.onTabClick(t)}>
-            <Kb.WithTooltip text={data[t].label} toastClassName="tab-tooltip">
+type State = {|
+  showingMenu: boolean,
+|}
+
+class TabBar extends React.PureComponent<Props, State> {
+  state = {showingMenu: false}
+
+  _attachmentRef = React.createRef<Kb.Box2>()
+  _getAttachmentRef = () => this._attachmentRef.current
+  _showMenu = () => this.setState({showingMenu: true})
+  _hideMenu = () => this.setState({showingMenu: false})
+  _menuHeader = () => ({
+    onClick: this.props.onProfileClick,
+    title: '',
+    view: (
+      <Kb.Box2 direction="vertical" gap="small" gapStart={true} gapEnd={true}>
+        <Kb.ConnectedNameWithIcon
+          username={this.props.username}
+          onClick={() => {
+            this._hideMenu()
+            this.props.onProfileClick()
+          }}
+          metaTwo={
+            <Kb.Text type="BodySmall" lineClamp={1}>
+              {this.props.fullname}
+            </Kb.Text>
+          }
+        />
+      </Kb.Box2>
+    ),
+  })
+  _menuItems = () => [
+    {
+      onClick: this.props.onProfileClick,
+      title: 'View Profile',
+    },
+    'Divider',
+    {
+      onClick: this.props.onSettings,
+      title: 'Settings',
+    },
+    {
+      onClick: this.props.onHelp,
+      title: 'Help',
+    },
+    {
+      danger: true,
+      onClick: this.props.onSignOut,
+      title: 'Sign out',
+    },
+  ]
+
+  render() {
+    const p = this.props
+    return (
+      !!p.username && (
+        <Kb.Box2 className="tab-container" direction="vertical" fullHeight={true}>
+          <Kb.Box2 direction="vertical" style={styles.header} fullWidth={true}>
+            <Kb.Box2 direction="horizontal" style={styles.osButtons} fullWidth={true} />
+            <Kb.ClickableBox onClick={this._showMenu}>
               <Kb.Box2
                 direction="horizontal"
+                gap="tiny"
                 fullWidth={true}
-                className={t === p.selectedTab ? 'tab-selected' : 'tab'}
-                style={styles.tab}
+                style={styles.nameContainer}
+                alignItems="center"
```

Comment:
`centerChildren={true}` to line this up
<img width="82" alt="image" src="https://user-images.githubusercontent.com/11968340/54610195-df7b6a80-4a2a-11e9-9639-2fd1f23c196c.png">

Reply:
tnx